objects/src/notes/script.rs

```diff
@@ -29,3 +29,37 @@ impl NoteScript {
         &self.code
     }
 }
+
+/// NoteScriptInputs holds the inputs that are placed on the stack at the initiation of note script
+/// execution.
+#[derive(Debug, Eq, PartialEq)]
+pub struct NoteScriptInputs {
+    inputs: Vec<Felt>,
+    hash: Digest,
+}
```

Comment:
minor nit: I'd change the doc comment to: "Holds the inputs which are placed onto the stack before a note's script is executed."

Also, I wonder if we should call it just `NoteInputs` and put it into a separate file.